src/scan/lexer_public.ts

```diff
@@ -616,11 +619,14 @@ export class Lexer {
 
                 // mode handling, must pop before pushing if a Token both acts as both
                 // otherwise it would be a NO-OP
+                // need to save the PUSH_MODE property as if the mode is popped
+                // patternIdxToPopMode is updated to reflect the new mode after popping the stack
+                let pushMode = patternIdxToPushMode[i]
```

Comment:
That's a fair point, and doesn't seem like nitpicking, since efficiency is one of the main goals of the project.  I'm glad I could contribute a temporary fix, as chevrotain's been great to use 👍